src/adapter/src/coord/peek.rs

```diff
@@ -492,6 +492,11 @@ impl crate::coord::Coordinator {
         // The peek is present on some specific compute instance.
         // Allow dataflow to cancel any pending peeks.
         if let Some(uuids) = self.client_pending_peeks.remove(&conn_id) {
+            self.metrics
```

Comment:
There are other times we do peek cancellation, see https://github.com/MaterializeInc/materialize/blob/24690cd41fd09e2fe0fb959054799d77a80c6355/src/pgwire/src/protocol.rs#L1506 for example.

If you need this metric for dataflow reasons, I suggest renaming it to `mz_canceled_dataflow_peeks_total` or something.

Reply:
`mz_canceled_dataflow_peeks_total` is also misleading, since not all peeks cause the creation of a dataflow. Fast-path peeks don't, and those are included in the metric too.

It's possible that we have different definitions of "peek" in mind. For me, a peek is a compute-level concept. Adapter asks compute to peek an arrangement. At the SQL layer we talk about queries and SELECTs, but not peeks, right? It seems to me that the code you linked handels cancelation of a query after the peek has already completed, which would mean a "canceled peeks" metric should not include these cancelations.

Reply:
Agree. I don't care too much what this metric's name is, as long as it's specific enough so we can add other metrics in the future in this space.